File: java/src/main/java/algorithm/sorting/BubbleSort.java
package algorithm.sorting;

import java.util.Arrays;

public class BubbleSort {
  static void bubbleSort(int[] arr) {
    for (int i = 0; i < arr.length; i++) {
      for (int j = 0; j < arr.length - i - 1; j++) {
        if (arr[j] > arr[j+1]) {
          int temp = arr[j];
          arr[j] = arr[j+1];
          arr[j+1] = temp;
        }
      }
    }
  }

  public static void main(String[] args) {
    int[] input = {1, 3, 7, 4, 2};
    Arrays.stream(input).forEach(System.out::print);
    bubbleSort(input);
    System.out.println();
    Arrays.stream(input).forEach(System.out::print);
  }
}

File: java/src/main/kotlin/algorithm/sorting/BubbleSort.kt
package kotlin.algorithm.sorting
fun bubbleSort(arr: IntArray) {
    for (i in 0..arr.size-1) {
        for(j in 0 .. arr.size-1-i) {
            if (arr[j] > arr[j+1]) {
                val temp = arr[j]
                arr[j] = arr[j+1]
                arr[j+1] = arr[j]
            }
        }
    }
}

fun main() {
    val array = intArrayOf(1, 3, 7, 4, 2)
    bubbleSort(array)
    println("Sorted: ${array.contentToString()}")
}
